### ISSUE TITLE
Issue With Locking Code

### ISSUE THREAD

**Issue by Philip-Evans** [FAXES/easycompile#10]
throw new Error('Invalid or incompatible cached data (cachedDataRejected)');
      ^

Error: Invalid or incompatible cached data (cachedDataRejected)
    at Module._extensions.<computed> [as .ProjectPhil] (C:\Users\Philip Evans\Desktop\ProjectPhil-Ticket_Bot\node_modules\easycompile\lib\index.js:46:13)
    at Module.load (node:internal/modules/cjs/loader:1091:32)
    at Module._load (node:internal/modules/cjs/loader:938:12)
    at Module.require (node:internal/modules/cjs/loader:1115:19)
    at require (node:internal/modules/helpers:130:18)
    at Object.<anonymous> (C:\Users\Philip Evans\Desktop\ProjectPhil-Ticket_Bot\index.js:10:1)
    at Module._compile (node:internal/modules/cjs/loader:1233:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1287:10)
    at Module.load (node:internal/modules/cjs/loader:1091:32)
    at Module._load (node:internal/modules/cjs/loader:938:12)

Node.js v20.5.1

**Comment by FAXES** [comment]
This error coms from the v8 engine from memory and can be caused by a syntax error in the code that was complied, or when a complied file is run on a different node version than what it complied.

Essentially if a file is complied on version 16.0.0 of Node.Js it can only run on Node version 16.0.0.

**Comment by Philip-Evans** [comment]
what do i change it to 20 ?

**Comment by Philip-Evans** [comment]
this is the code 
```const originalEmit = process.emit, v8 = require("v8");
v8.setFlagsFromString('--no-lazy');
if (Number.parseInt(process.versions.node, 10) >= 12) v8.setFlagsFromString('--no-flush-bytecode');

process.emit = function (name, data) {
    if(name.toLowerCase() !== "warning" && typeof data !== "object" && data.name !== "ExperimentalWarning") return originalEmit.apply(process, arguments)
}

require("easycompile").changeExtension(".ProjectPhil")
require("./src/client/client").start()```

**Comment by FAXES** [comment]
> what do i change it to 20 ?

Not sure what you mean by this, but as long as it's complied and run on the same node version it should be fine.

**Comment by Philip-Evans** [comment]
it is on petro its ran on v20 and my pc on v20

**Comment by Philip-Evans** [comment]
doesnt run on my pc to

**Comment by Philip-Evans** [comment]
@FAXES

**Comment by Philip-Evans** [comment]
issue script not working for directorys

**Comment by FAXES** [comment]
> it is on petro its ran on v20 and my pc on v20

Must be the exact version down to the minor and fix releases, so if it's compiled on 20.1.2 it must be run on 20.1.2 as far as I know.

Directories should work, I personally don't use the script though. I know @braxtongpoll does though.

**Comment by braxtongpoll** [comment]
@ProjectPhil , @FAXES is correct I due use this script just a slightly modified version. Directories work without issue and from our experience the node version be the minor fix release. As in your example v20.5.1 instead of v20.5 or v20
